### PR TITLE
feat(license): Listing obligations by license

### PR DESCRIPTION
**Pull request by tuannn2**
[//]: # (Copyright Bosch.IO GmbH 2020)
[//]: # (This program and the accompanying materials are made)
[//]: # (available under the terms of the Eclipse Public License 2.0)
[//]: # (which is available at https://www.eclipse.org/legal/epl-2.0/)
[//]: # (SPDX-License-Identifier: EPL-2.0)

# Add new API Listing obligations by license

### How To Test?
- Run command
```
$ curl 'https://sw360.org/api/licenses/MIT/obligations' -i -X GET \
    -H 'Authorization: Bearer eyJhbGciOiJSUzI1NiIsInR5cCI6IkpXVCJ9.eyJhdWQiOlsic3czNjAtUkVTVC1BUEkiXSwidXNlcl9uYW1lIjoiYWRtaW5Ac3czNjAub3JnIiwic2NvcGUiOlsiYWxsIl0sImV4cCI6MTcwMjM1MjAyNSwiYXV0aG9yaXRpZXMiOlsiUkVBRCIsIldSSVRFIl0sImp0aSI6IkxxSGFjbU5QVThlcWlEMnNpVTFlWGZ3bHB3USIsImNsaWVudF9pZCI6InRydXN0ZWQtc3czNjAtY2xpZW50In0.gadnfJIwnx4D2a77wXBkZCUKldzOTnQNul4v2slHSkAePjTe7HqA4ktRQ4UoWWeb_7XRy2kvfZIEkc4MmiN7mIxKIB2PYNPHyPiDy6EO9VJPO5_TqWfdXGOagGuefHowXNLeB54W7N-42kD5OLgh_xgx0CXV7mi7m2fs32ysnwQWR2B0xUC_CAIqp9PZQJK062KW01tec2c6vM8_yeNZTP4y4NKw-jqe9LpH8flpzxOtX8O7iVkqcZ4rFwcl6vouSZ5YLL7Ru6Kh-NOmvmHosHyCUT6LHbiSyid4kKNK8Ja_mK1CdMDWmTa76ua59OJC-PbTbmv4pkhVll6QNh2RFQ' \
    -H 'Accept: application/hal+json'
```

- Output 
```
{
  "_embedded" : {
    "sw360:obligations" : [ {
      "text" : "This is text of Obligation 1",
      "title" : "Obligation 1",
      "obligationType" : "PERMISSION",
      "_links" : {
        "self" : {
          "href" : "https://sw360.org/api/obligations/0001"
        }
      }
    }, {
      "text" : "This is text of Obligation 2",
      "title" : "Obligation 2",
      "obligationType" : "OBLIGATION",
      "_links" : {
        "self" : {
          "href" : "https://sw360.org/api/obligations/0002"
        }
      }
    } ]
  },
  "_links" : {
    "curies" : [ {
      "href" : "https://sw360.org/docs/{rel}.html",
      "name" : "sw360",
      "templated" : true
    } ]
  }
}
```

### Checklist
Must:
- [ ] All related issues are referenced in commit messages and in PR

### CODE REVIEW

**Comment by akshitjoshii** [comment]
Testing was successful
2024-01-02 10_53_50-SW360_NEW  Running  - Oracle VM VirtualBox

File: rest/resource-server/src/test/java/org/eclipse/sw360/rest/resourceserver/restdocs/LicenseSpecTest.java [hunk 1]
```diff
@@ -197,6 +203,25 @@ public void should_document_get_license() throws Exception {
...
+                .header("Authorization", "Bearer " + accessToken)
+                .accept(MediaTypes.HAL_JSON))
+                .andExpect(status().isOk())
+                .andDo(this.documentationHandler.document(
+                        links(
+                                linkWithRel("curies").description("Curies are used for online documentation")
+                        ),
+                        responseFields(
+                                subsectionWithPath("_embedded.sw360:obligations[]title").description("The title of the obligation"),
+                                subsectionWithPath("_embedded.sw360:obligations[]obligationType").description("The obligationType of the obligation"),
```
Comment (afsahsyeda) [inline rest/resource-server/src/test/java/org/eclipse/sw360/rest/resourceserver/restdocs/LicenseSpecTest.java:219]: I believe this should be "The type of the obligation" or "Obligation type" instead.

File: backend/src/src-licenses/src/main/java/org/eclipse/sw360/licenses/db/LicenseDatabaseHandler.java [hunk 2]
```diff
@@ -849,6 +849,12 @@ public List<Obligation> getObligationsByIds(Collection<String> ids) {
         return obligations;
     }
 
+    public List<Obligation> getObligationsByLicenseId(String id) throws SW360Exception {
+        License license = getLicenseForOrganisation(id, "?");
```
Comment (afsahsyeda) [inline backend/src/src-licenses/src/main/java/org/eclipse/sw360/licenses/db/LicenseDatabaseHandler.java:853]: What is the usage of passing "?" as the organization?

**Review by afsahsyeda — CHANGES_REQUESTED** [review]

File: rest/resource-server/src/test/java/org/eclipse/sw360/rest/resourceserver/restdocs/LicenseSpecTest.java (hunk 1 above)
Comment (tuannn2) [reply rest/resource-server/src/test/java/org/eclipse/sw360/rest/resourceserver/restdocs/LicenseSpecTest.java]: fixed.

**Review by afsahsyeda — APPROVED** [review]

**Comment by afsahsyeda** [comment]
Changes look good to me.

**Pull request merged**